api/constants/constants.go

```diff
@@ -148,25 +148,38 @@ const (
 	// SSHRSAType is the string which specifies an "ssh-rsa" formatted keypair
 	SSHRSAType = "ssh-rsa"
 
-	// OktaAssignmentStatusPending is represents a pending status for an Okta assignment.
+	// OktaAssignmentStatusPending represents a pending status for an Okta assignment.
 	OktaAssignmentStatusPending = "pending"
 
-	// OktaAssignmentStatusProcessing is represents an Okta assignment which is currently being acted on.
+	// OktaAssignmentStatusProcessing represents an Okta assignment which is currently being acted on.
 	OktaAssignmentStatusProcessing = "processing"
 
-	// OktaAssignmentStatusSuccessful is represents a successfully applied Okta assignment.
+	// OktaAssignmentStatusSuccessful represents a successfully applied Okta assignment.
 	OktaAssignmentStatusSuccessful = "successful"
 
-	// OktaAssignmentStatusFailed is represents an Okta assignment which failed to apply. It will be retried.
+	// OktaAssignmentAccessListRemovalPendingOktaCleanup indicates that the user has been removed
+	// from the Access List, but the corresponding cleanup action—removal from
+	// the associated Okta group—has not yet been executed.
+	//
+	// This status represents a transitional state where internal access control
+	// has been updated (i.e., the user was removed from the Access List), but the
+	// change is not yet reflected in the user assignments state. This prevents
+	// importing-recreating Okta assigns in teleport  during asynchronous Okta ACL sync.
+	// when the ongoing Remove user from Access List operation is in progress.
+	// Once the user assignments are evaluator successfully processes the Access List
+	// membership changes, the associated assignment will be deleted.
+	OktaAssignmentAccessListRemovalPendingOktaCleanup = "acl_removal_pending_okta_cleanup"
```

Comment:
I wonder if we can't come up with a more generic reusable name for this sort of situation where we don't want delete assignments scheduled for cleanup? Maybe "locked" or "cleanup_locked"? Or you think it must be bound to ACLs in particular?

Reply:
I think the status should describe _what_ is going on and not _when_ it may happen. That locks it to be used only in a certain scenario. I think we should find what is the actual transition on a higher level.

How about "cleanup_processing" (that would correspond to "processing" which we already have)? That would make sense to me that when the status is "cleanup_processing" the user can't be added back to Teleport ACL (or anything else) from Okta because Teleport haven't removed the user from Okta yet.